deps/build.jl

```diff
@@ -5,58 +5,59 @@ using Compat
 group = library_group("symengine")
 
 deps = [
-    libgmp = library_dependency("libgmp", group = group),
-    libsymengine = library_dependency("libsymengine", depends = [libgmp])
+    gmp = library_dependency("gmp", aliases=["libgmp", "gmp", "mpir", "libmpir"], group = group),
+    mpfr = library_dependency("mpfr", aliases=["libmpfr", "mpfr"], group=group, depends=[gmp]),
+    mpc = library_dependency("mpc", aliases=["libmpc", "mpc"], group=group, depends=[mpfr]),
+    symengine = library_dependency("symengine", aliases=["libsymengine", "symengine"], depends = [mpc])
 ]
 
-prefix=joinpath(BinDeps.depsdir(libsymengine), "usr")
-
-provides(AptGet,
-    @compat Dict(
-        "libgmp-dev" => [libgmp]
-    ))
-
-provides(Yum,
-    @compat Dict(
-        "gmp-devel" => [libgmp]
-    ))
-
-@osx_only begin
-    if Pkg.installed("Homebrew") === nothing
-        print("Homebrew package not installed, please run Pkg.add(\"Homebrew\") to use it to download dependencies")
-    else
-        using Homebrew
-        provides(Homebrew.HB, "gmp", [libgmp], os = :Darwin)
-    end
+prefix=joinpath(BinDeps.depsdir(symengine), "usr")
+
+xx(t...) = (OS_NAME == :Windows ? t[1] : (OS_NAME == :Linux || length(t) == 2) ? t[2] : t[3])
```

Comment:
It's a simple function to do different things for different OSs. It's used below. (Copied it from here, https://github.com/JuliaGraphics/Cairo.jl/blob/master/deps/build.jl#L102)